pom.xml

```diff
@@ -278,6 +278,7 @@
       </activation>
       <properties>
         <java.version>1.8</java.version>
+        <iceberg.version>1.5.0</iceberg.version>
```

Comment:
does it works well on ch backends? For velox backends, the iceberg version must use 1.8.0

Reply:
For Velox backend tests, we already upgraded to JDK17. I think the intention here is to enable JDK8 + Spark-355

Reply:
If we try to build with JD8 and Spark 355, iceberg should not be included. The changes here don't seem necessary to me.
Besides, we need to define `iceberg.version` in spark config file instead of jdk config file so that we can use different iceberg versions for different sparks.

Reply:
If Iceberg and Spark versions are bound together, it means that after upgrading Spark to version 3.5.5, it will no longer be possible to use JDK 8. However, Spark 3.5.5 itself still supports the use of JDK 8.
Use different versions of JDK and Spark to support different versions of Iceberg.

Reply:
Not get your mind. I have runed Iceberg 1.5.0 with Spark 3.5.5 before, and it can't work that way. Thus If we want use Spark 3.5.5 with JDK8, then iceberg can not be involved.

Reply:
@jackylee-ch 
I made a quick patch to test iceberg 1.5 + Spark-355 + JDK8, 
https://github.com/apache/incubator-gluten/pull/9001/

and it looks like all the unit tests passed:
https://github.com/apache/incubator-gluten/actions/runs/13859404046/job/38785108628

The JDK17 based tests failed as expected.
https://github.com/apache/incubator-gluten/actions/runs/13859404046/job/38785108162

Reply:
> If we try to build with JD8 and Spark 355, iceberg should not be included. The changes here don't seem necessary to me. Besides, we need to define `iceberg.version` in spark config file instead of jdk config file so that we can use different iceberg versions for different sparks.

+1 
it seems we should not bind JDK version with iceberg version as this will impact with Spark-344 also.

Reply:
> @jackylee-ch I made a quick patch to test iceberg 1.5 + Spark-355 + JDK8, #9001
> 
> and it looks like all the unit tests passed: https://github.com/apache/incubator-gluten/actions/runs/13859404046/job/38785108628

It seems you were using https://github.com/apache/incubator-gluten/pull/8890 to test it, sorry for that as I have removed the `-Piceberg` from spark-3.5.5 and jdk8 GA.

BTW, current PR has already met the [problem](https://github.com/apache/iceberg/pull/11731#issuecomment-2530114142) I described before, the vanilla spark would core dump in `VeloxTPCHIcebergSuite`
https://github.com/apache/incubator-gluten/actions/runs/13916861429/job/38941504564?pr=9031

Reply:
@jackylee-ch Thanks for point out, I didnt realize this change - so it looks like Velox backend does not support this combination `JDK8 + Spark355 + Iceberg1.5`

Not sure if this applies to the CK backend also. In case CK backend can work with this, could we add a new special profile so it wont impact Velox backend?

Cc: @baibaichen

Reply:
Unfortunately, this core dump issue occurs in the vanilla Spark 3.5.5 environment with Iceberg 1.5.0, and it is unrelated to the Velox backend. Similar problems can also be encountered when using the CH backend.

Reply:
> > If we try to build with JD8 and Spark 355, iceberg should not be included. The changes here don't seem necessary to me. Besides, we need to define `iceberg.version` in spark config file instead of jdk config file so that we can use different iceberg versions for different sparks.
> 
> +1 it seems we should not bind JDK version with iceberg version as this will impact with Spark-344 also.

en, i lose of that.  Spark-344 is another version of Iceberg